src/topology/config/watcher.rs

```diff
@@ -0,0 +1,131 @@
+use crate::Error;
+use notify::{raw_watcher, Op, RawEvent, RecommendedWatcher, RecursiveMode, Watcher};
+use std::{
+    path::{Path, PathBuf},
+    sync::mpsc::{channel, Receiver},
+    thread,
+    time::Duration,
+};
+
+/// Per notify own documentation, it's advised to have delay of more than 30 sec,
+/// so to avoid receiving repetitions of previous events on macOS.
+///
+/// But, config and topology reload logic can handle:
+///  - Invalid config, caused either by user or by data race.
+///  - Frequent changes, caused by user/editor modifying/saving file in small chunks.
+/// so we can use smaller, more responsive delay.
+pub const CONFIG_WATCH_DELAY: std::time::Duration = std::time::Duration::from_secs(1);
```

Comment:
This seems like it would be a valuable configuration option.

Reply:
For us yes, but not so much for the user as the benefits/flexibility are small. At best user could either decrease lag by max 1 second, but in turn cause vector to reconfigure it self feverishly on every write, or increase lag by larger amount and gain some small unnoticeable benefits in some cases.